Error in CI with pip upgrade

We're seeing an error in our CI runs in windows with the following error:

Run pip install  --user --upgrade pip
Requirement already satisfied: pip in c:\hostedtoolcache\windows\python\3.8.10\x64\lib\site-packages (22.1.1)
Collecting pip
  Downloading pip-22.1.2-py3-none-any.whl (2.1 MB)
     ---------------------------------------- 2.1/2.1 MB 66.6 MB/s eta 0:00:00
ERROR: To modify pip, please run the following command:
c:\hostedtoolcache\windows\python\3.8.10\x64\python.exe -m pip install --user --upgrade pip
WARNING: There was an error checking the latest version of pip.
Error: Process completed with exit code 1.

--- a/.changes/unreleased/Under the Hood-20220601-105245.yaml
+++ b/.changes/unreleased/Under the Hood-20220601-105245.yaml
@@ -1,0 +1,7 @@
+kind: Under the Hood
+body: Fix pip upgrade step in CI for Windows
+time: 2022-06-01T10:52:45.872931-04:00
+custom:
+  Author: gshank
+  Issue: "5321"
+  PR: "5320"

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -48,14 +48,14 @@ jobs:
 
       - name: Install python dependencies
         run: |
-          pip install --user --upgrade pip
-          pip --version
-          pip install pre-commit
+          python -m pip install --user --upgrade pip
+          python -m pip --version
+          python -m pip install pre-commit
           pre-commit --version
-          pip install mypy==0.942
+          python -m pip install mypy==0.942
           mypy --version
-          pip install -r requirements.txt
-          pip install -r dev-requirements.txt
+          python -m pip install -r requirements.txt
+          python -m pip install -r dev-requirements.txt
           dbt --version
 
       - name: Run pre-commit hooks
@@ -69,7 +69,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10']
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
 
     env:
       TOXENV: "unit"
@@ -86,9 +86,9 @@ jobs:
 
       - name: Install python dependencies
         run: |
-          pip install --user --upgrade pip
-          pip --version
-          pip install tox
+          python -m pip install --user --upgrade pip
+          python -m pip --version
+          python -m pip install tox
           tox --version
 
       - name: Run tox
@@ -113,7 +113,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10']
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
         os: [ubuntu-latest]
         include:
           - python-version: 3.8
@@ -149,9 +149,9 @@ jobs:
 
       - name: Install python tools
         run: |
-          pip install --user --upgrade pip
-          pip --version
-          pip install tox
+          python -m pip install --user --upgrade pip
+          python -m pip --version
+          python -m pip install tox
           tox --version
 
       - name: Run tests
@@ -190,9 +190,9 @@ jobs:
 
       - name: Install python dependencies
         run: |
-          pip install --user --upgrade pip
-          pip install --upgrade setuptools wheel twine check-wheel-contents
-          pip --version
+          python -m pip install --user --upgrade pip
+          python -m pip install --upgrade setuptools wheel twine check-wheel-contents
+          python -m pip --version
 
       - name: Build distributions
         run: ./scripts/build-dist.sh
@@ -210,7 +210,7 @@ jobs:
 
       - name: Install wheel distributions
         run: |
-          find ./dist/*.whl -maxdepth 1 -type f | xargs pip install --force-reinstall --find-links=dist/
+          find ./dist/*.whl -maxdepth 1 -type f | xargs python -m pip install --force-reinstall --find-links=dist/
 
       - name: Check wheel distributions
         run: |
@@ -219,7 +219,7 @@ jobs:
       - name: Install source distributions
         # ignore dbt-1.0.0, which intentionally raises an error when installed from source
         run: |
-          find ./dist/dbt-[a-z]*.gz -maxdepth 1 -type f | xargs pip install --force-reinstall --find-links=dist/
+          find ./dist/dbt-[a-z]*.gz -maxdepth 1 -type f | xargs python -m pip install --force-reinstall --find-links=dist/
 
       - name: Check source distributions
         run: |

--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ endif
 .PHONY: dev
 dev: ## Installs dbt-* packages in develop mode along with development dependencies.
 	@\
-	pip install -r dev-requirements.txt -r editable-requirements.txt && \
+	pip install -r dev-requirements.txt -r editable-requirements.txt
 
 .PHONY: mypy
 mypy: .env ## Runs mypy against staged changes for static type checking.

